repos/tap/components/list/treeList/treeList.js

```diff
@@ -0,0 +1,111 @@
+import React, { useCallback } from 'react'
+import { pickKeys, noPropArr } from '@keg-hub/jsutils'
+import { useTheme } from '@keg-hub/re-theme'
+import { useSelector, shallowEqual } from 'react-redux'
+import { setFeatureActive } from 'SVActions/features'
+import { loadFeature } from 'SVActions/features/loadFeature'
+import {
+  View,
+  Loading,
+  Text,
+} from 'SVComponents'
+import { Values } from 'SVConstants'
+import TreeView from 'react-native-final-tree-view'
+import { ChevronDown } from 'SVAssets/icons'
+import { useStoreItems } from 'SVHooks/store/useStoreItems'
+
+const { CATEGORIES } = Values
+
+/**
+ * Finds the node based on given id
+ * @param {string} id 
+ * @param {Array<object>} nodes 
+ * 
+ * @returns {object} node
+ */
+const findNode = (id, nodes) => {
+  let foundNode = {}
+  nodes.reduce((__, node) => {
+    if (node.id === id) foundNode = node
+    if (node.children.length > 0) {
+      findNode(id, node.children)
+    }
+  }, {})
+
+  return foundNode
+}
+
+/**
+ * TreeList
+ * @param {Object} props 
+ */
+export const TreeList = props => {
+
+  const { features, fileTree=noPropArr } = useStoreItems([CATEGORIES.FEATURES, CATEGORIES.FILE_TREE])
+
+  const onItemPress = useCallback(({node}) => {
+    // for now only supporting feature file/content
+    const match = features.find(feature => feature.fullPath === node.fullPath)
+    match && loadFeature(match)
+  }, [ features, setFeatureActive ])
+  
+  const getCollapsedNodeHeight = useCallback(({id}) => {
+    const node = findNode(id, fileTree)
+    return (node?.children?.length === 0 && node?.type === 'folder')
```

Comment:
This line => `node?.children?.length === 0 && node?.type === 'folder'` is used in a number of places. Makes me think it should be extracted into a utility method

Reply:
will move to util `isEmptyFolderNode`.  i was going to go with `isEmptyFolder`, but that feels too general and this method it's tied to the file tree node.
but I'm open to different names if you have suggestions